lib/tst_test.c

```diff
@@ -629,7 +637,7 @@ static void do_cleanup(void)
 	if (tst_test->needs_device && tdev.dev)
 		tst_release_device(tdev.dev);
 
-	if (needs_tmpdir() && tmpdir_created) {
+	if (tst_tmpdir_created()) {
 		/* avoid munmap() on wrong pointer in tst_rmdir() */
 		tst_futexes = NULL;
 		tst_rmdir();
```

Comment:
Looks OK at first glance. I will have to have a second look and do a few tests later but it's most likely correct.